src/MediaWiki.ts

```diff
@@ -85,12 +185,17 @@ class MediaWiki {
         },
         method: 'POST',
       })
-        .then((resp) => {
+        .then(async (resp) => {
           if (resp.data.login.result !== 'Success') {
             throw new Error('Login Failed')
           }
 
+          /*
+           TODO: Cookie is shared between Downloader and Mediawiki, probably antipattern. Use as interim solution for now.
+           Also, double-check possible race condition - cookies should be set before checking capabilities.
+           */
           downloader.loginCookie = resp.headers['set-cookie'].join(';')
```

Comment:
Indee... here would help to clearly define what is the duty of the `Mediawiki` and `Downloader` class. Once this is clear, easier to refactor.